Adds OpenFaaS cloud community secrets documentation

ewilde opened the pull request:
Adds more documentation for OpenFaaS Cloud Community secrets

rgee0 commented on docs/openfaas-cloud/secrets.md:
```diff
@@ -0,0 +1,116 @@
+
+To consume a **secret** in a function:
+
+1.\ Create the secret object `secret.yml` using literals or from a file
```
The following command will result in a file names `secrets.yml` rather than `secret.yml`

rgee0 commented on docs/openfaas-cloud/secrets.md:
```diff
@@ -0,0 +1,116 @@
+> **Note:** GitHub `{username}` prefix is removed in the `stack.yml`
+
+
+3.\ Update you function code to read the secret from `/var/openfaas/secrets/{key}`
```
Update you => Update your

ewilde replied:
👍 corrected

rgee0 commented on docs/openfaas-cloud/secrets.md:
```diff
@@ -0,0 +1,116 @@
+> **Note:** GitHub `{username}` prefix is removed in the `stack.yml`
+
+
+3.\ Update you function code to read the secret from `/var/openfaas/secrets/{key}`
```
Should we add that for Go functions the OFC SDK offers `ReadSecret()` to enable functions to read secrets?  Perhaps as a note.

ewilde replied:
👍 good idea

rgee0 commented on docs/openfaas-cloud/secrets.md:
```diff
@@ -0,0 +1,116 @@
+- `/var/openfaas/secrets/username`
+- `/var/openfaas/secrets/password` 
+
+4.\ Commit `secret.yml` and other modified files. **Note** `secret.yml` must be in the root of your GitHub repository
```
secret.yml => secrets.yml

ewilde replied:
👍 corrected

rgee0 commented on docs/openfaas-cloud/secrets.md:
````diff
@@ -0,0 +1,116 @@
+
+Let's write the example function:
+
+```yaml
````
yaml => python

alexellis commented:
@ewilde please see comments

ewilde commented:
@rgee0 thank you for the review, hopefully i've addressed the comments. I've removed the older example as it does, as you mentioned, repeat what is covered here.

alexellis commented on docs/openfaas-cloud/self-hosted/secrets.md:
```diff
@@ -30,86 +30,4 @@ The preferred method available for Kubernetes uses SealedSecrets by Bitnami. We
-
-If in doubt check your results against [this reference repository](https://github.com/alexellis/my-fn).
-
+See the community [secrets](../secrets.md) documentation for information on how to use sealed secrets. 
```
Not specifically about "community". I'd remove

ewilde replied:
👍

alexellis commented on mkdocs.yml:
```diff
@@ -135,6 +135,7 @@ nav:
     - YAML: ./reference/yaml.md
   - OpenFaaS Cloud:
       - Intro: ./openfaas-cloud/intro.md
+      - Secrets: ./openfaas-cloud/secrets.md
```
Secret management?

ewilde replied:
how about  `manage secrets` like here ? https://docs.openfaas.com/cli/secrets/

alexellis commented on docs/openfaas-cloud/secrets.md:
```diff
@@ -0,0 +1,45 @@
+OpenFaaS Cloud **secret** objects lets you store and manage sensitive information securely
```
Why do we have secret in bold? There's not a thing called a secret object

ewilde replied:
👍

alexellis commented on docs/openfaas-cloud/secrets.md:
```diff
@@ -0,0 +1,45 @@
+OpenFaaS Cloud **secret** objects lets you store and manage sensitive information securely
+
+
+## Using secrets
```
Please can you follow the guidelines for headings in the imperative?

The previous heading was Seal a secret / Seal secrets. We need to maintain this terminology

ewilde replied:
👍

alexellis commented on docs/openfaas-cloud/secrets.md:
```diff
@@ -0,0 +1,45 @@
+
+Secrets in OpenFaaS Cloud will be mounted as data volumes and made accessible to your function
+
+### Creating secrets
```
Define a secret

ewilde replied:
Should this be `Define a sealed secret` following your recommendation to above?

alexellis commented on docs/openfaas-cloud/secrets.md:
```diff
@@ -0,0 +1,45 @@
+
+To consume a **secret** in a function:
+
+1.\ Create the secret object `secrets.yml` using literals or from a file
```
This is not a secret object. It's a SealedSecret

ewilde replied:
@alexellis isn't it a kubernetes secret object by the time you consume it in your function?

alexellis commented on docs/openfaas-cloud/secrets.md:
```diff
@@ -0,0 +1,45 @@
+
+If you are writing your code in Go, the OpenFaaS Cloud SDK provides a [ReadSecret](https://github.com/openfaas/openfaas-cloud/blob/master/sdk/secrets.go) method.
+
+4.\ Commit `secrets.yml` and other modified files. **Note** `secrets.yml` must be in the root of your GitHub repository
```
What does the backslash do after 4.?

alexellis replied:
The secrets.yml file should be placed at the / in the root of

ewilde replied:
> The secrets.yml file should be placed at the / in the root of

Don't understand that re-wording?

ewilde replied:
> What does the backslash do after 4.?

markdown escape to stop auto numbering of lists, it would reset them to 1. because of the code block between items

alexellis commented on docs/openfaas-cloud/secrets.md:
```diff
@@ -0,0 +1,45 @@
+- `/var/openfaas/secrets/username`
+- `/var/openfaas/secrets/password` 
+
+If you are writing your code in Go, the OpenFaaS Cloud SDK provides a [ReadSecret](https://github.com/openfaas/openfaas-cloud/blob/master/sdk/secrets.go) method.
```
Where is the troubleshooting section?

ewilde replied:
added back

alexellis commented:
Please can we also link to the SealedSecrets repo?

alexellis commented:
This documentation was already present and if possible I'd like to improve it rather than re-write it wholesale. I noticed some of the terminology was out of date and so I've made some changes.

https://docs.openfaas.com/openfaas-cloud/secrets/

Is this clearer now Ed?

Which parts are still missing / unclear?

ewilde commented:
lgtm
